1.train_models/log_reg_plate_1_cp_fs_data/nbconverted/train_model.py

```diff
@@ -0,0 +1,237 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Training a Logistic Regression Model
+
+# ## Imports
+
+# In[ ]:
+
+
+import pandas as pd
+import numpy as np
+from pathlib import Path
+from sklearn.preprocessing import LabelEncoder, OneHotEncoder
+from sklearn.linear_model import LogisticRegression
+
+from sklearn.metrics import (
+    confusion_matrix,
+    ConfusionMatrixDisplay,
+    precision_score,
+    accuracy_score,
+    classification_report,
+)
+from sklearn.model_selection import train_test_split, KFold
+
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+from joblib import dump, load
+
+
+# ## Find the git root Directory
+
+# In[ ]:
+
+
+# Get the current working directory
+cwd = Path.cwd()
+
+if (cwd / ".git").is_dir():
+    root_dir = cwd
+
+else:
+    root_dir = None
+    for parent in cwd.parents:
+        if (parent / ".git").is_dir():
+            root_dir = parent
+            break
+
+# Check if a Git root directory was found
+if root_dir is None:
+    raise FileNotFoundError("No Git root directory found.")
+
+
+# # Seed and Generator for Reproducibility
+
+# In[ ]:
+
+
+rnd_val = 0  # Random value for all seeds
+rng = np.random.default_rng(seed=rnd_val)  # random number generator
+
+
+# ## Create paths
+
+# In[ ]:
+
+
+filename1 = "Plate_1_sc_norm_fs.parquet"
+plate_path = Path(
+    f"{root_dir}/nf1_painting_repo/3.processing_features/data/feature_selected_data"
+)
+
+path1 = plate_path / filename1
+
+data_path = Path("data")
+
+# Create the parent directories if they don't exist
+data_path.mkdir(parents=True, exist_ok=True)
+
+
+# ## Generate plate dataframe
+
+# In[ ]:
+
+
+# Returns the dataframe generated by the csv
+platedf = pd.read_parquet(path1)
+
+
+# # Preprocess Data
+
+# ## Create Classes
+
+# In[ ]:
+
+
+target_column = "Metadata_genotype"
+stratify_column = "Metadata_Well"
+
+# These represent the fractions of the entire dataset
+train_val_frac = 0.85
+test_frac = 1 - train_val_frac
+val_frac = 0.15
```

Comment:
Is this expected to add to 1? (train + test + val)

Reply:
The val_frac and the test_frac aren't necessary for this analysis since I'm using kfold. This is fixed now.